### ISSUE TITLE
Generative Pre-Training for Speech with Autoregressive Predictive Coding

### ISSUE THREAD

**Issue by juanmc2005** [juanmc2005/phd-bib#33]
## [Generative Pre-Training for Speech with Autoregressive Predictive Coding](https://arxiv.org/abs/1910.12607v2)
*Learning meaningful and general representations from unannotated speech that are applicable to a wide range of tasks remains challenging. In this paper we propose to use autoregressive predictive coding (APC), a recently proposed self-supervised objective, as a generative pre-training approach for learning meaningful, non-specific, and transferable speech representations. We pre-train APC on large-scale unlabeled data and conduct transfer learning experiments on three speech applications that require different information about speech characteristics to perform well: speech recognition, speech translation, and speaker identification. Extensive experiments show that APC not only outperforms surface features (e.g., log Mel spectrograms) and other popular representation learning methods on all three tasks, but is also effective at reducing downstream labeled data size and model parameters. We also investigate the use of Transformers for modeling APC and find it superior to RNNs.*

### Cite this paper
```bibtex
@inproceedings{chung2020generative,
  title = {Generative pre-training for speech with autoregressive predictive coding},
  author = {Chung, Yu-An and Glass, James},
  booktitle = {ICASSP},
  year = {2020}
}
```

**Comment by juanmc2005** [comment]
### Description
In this paper, the authors apply the self-supervised Autoregressive Predictive Coding method (APC) to learn general speech representations for transfer. They evaluate their approach using two different architectures: a RNN of GRU cells and a Transformer, and on three different tasks: speech recognition, speech translation and speaker identification.
They also compare their model to Contrastive Predictive Coding (CPC), Problem-Agnostic Speech Encoder (PASE) and log Mel spectrograms.

### APC
Autoregressive predictive coding is a method inspired from natural language processing architectures, in which the model learns by predicting the next segment of audio in the sequence by using the L1 loss function (absolute value of the difference between target and prediction).
Ground truth is easy to obtain by shifting the sequence 1 step to the right, which makes this method an attractive approach for self-supervised learning.

### Results
On all transfer experiments, they found transformers to perform considerably better than the RNN.

Experiments on speech recognition show that freezing weights during transfer actually benefit learning and obtain better performance, which is quite counterintuitive, but no further analysis on this aspect is shown.

Another interesting result is that in all experiments, APC outperforms log Mel spectrograms, also showing a more effective cost-efficiency balance in terms of the number of parameters in the model. A transformer with APC can obtain the same results as log Mel spectrograms with half of the layers for fine-tuning, as observed in the speech recognition experiments.

### Takeaway
Overall this can be an interesting direction as it can bring NLP and speech architectures closer together. A continual variant of APC could be developed by taking advantage of the temporal nature of the model, maybe past sequences could help decode the next step of a current sequence, or just part of it. This can be worth exploring in the future.
